client/state/user-settings/reducer.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+import { merge, omit } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import {
+	USER_SETTINGS_FETCH,
+	USER_SETTINGS_FETCH_SUCCESS,
+	USER_SETTINGS_FETCH_FAILURE,
+	USER_SETTINGS_SAVE_SUCCESS,
+	USER_SETTINGS_RECEIVE,
+	USER_SETTINGS_UNSAVED_SET,
+	USER_SETTINGS_UNSAVED_REMOVE,
+} from 'state/action-types';
+import { createReducer } from 'state/utils';
+import { decodeEntities } from 'lib/formatting';
+
+/*
+ * Decodes entities in those specific user settings properties
+ * that the REST API returns already HTML-encoded
+ */
+function decodeUserSettingsEntities( data ) {
+	const decodedValues = {
+		display_name: data.display_name && decodeEntities( data.display_name ),
+		description: data.description && decodeEntities( data.description ),
+		user_URL: data.user_URL && decodeEntities( data.user_URL )
+	};
+
+	return merge( {}, data, decodedValues );
+}
+
+/**
+ * Returns the updated remote values state after an action has been dispatched.
+ * The remote values state reflects preferences which are persisted to the REST
+ * API current user settings endpoint.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export const settings = createReducer( false, {
+	[ USER_SETTINGS_RECEIVE ]: ( state, { values } ) => {
+		return merge( {}, state || {}, decodeUserSettingsEntities( values ) );
+	}
+} );
```

Comment:
> I use createReducer because it provides a nice API where I pass a map actionType -> handler instead of creating a big switch statement.

as a first observation here, we wouldn't be dealing with any big `switch` statement because we're only listening to a single action type. all of this work is just complicating what could be a single simple conditional.

> it seems to be used everywhere else. Or is the preferred style to use the good old switch?

well, the premise is a false dichotomy but yes it's often used but not everywhere. I encourage people not to jump to `createReducer` unless it brings some specific benefit they want because it adds overhead and complexity to the code. more likely than not, `createReducer` doesn't do what we think it does, which is difficult for maintaining code that uses it. it's also _not_ used everywhere, as plenty of reducers are simply built to their needs.

```js
export const settings = ( state = null, { type, values } ) =>
	USER_SETTINGS_RECEIVE === type
		? { ...state, ...decodeUserSettingsEntities( values ) }
		: state;
```